functions/imagemagick/index.js

```diff
@@ -30,115 +30,94 @@ const {BLURRED_BUCKET_NAME} = process.env;
 
 // [START functions_imagemagick_analyze]
 // Blurs uploaded images that are flagged as Adult or Violence.
-exports.blurOffensiveImages = event => {
-  const object = event.data || event; // Node 6: event.data === Node 8+: event
-
-  // Exit if this is a deletion or a deploy event.
-  if (object.resourceState === 'not_exists') {
-    console.log('This is a deletion event.');
-    return;
-  } else if (!object.name) {
-    console.log('This is a deploy event.');
-    return;
-  }
+exports.blurOffensiveImages = async event => {
+  const object = event;
 
   const file = storage.bucket(object.bucket).file(object.name);
   const filePath = `gs://${object.bucket}/${object.name}`;
 
-  // Ignore already-blurred files (to prevent re-invoking this function)
-  if (file.name.startsWith('blurred-')) {
-    console.log(`The image ${file.name} is already blurred.`);
-    return;
-  }
-
   console.log(`Analyzing ${file.name}.`);
 
-  return client
-    .safeSearchDetection(filePath)
-    .catch(err => {
-      console.error(`Failed to analyze ${file.name}.`, err);
-      return Promise.reject(err);
-    })
-    .then(([result]) => {
-      const detections = result.safeSearchAnnotation;
-
-      if (
-        detections.adult === 'VERY_LIKELY' ||
-        detections.violence === 'VERY_LIKELY'
-      ) {
-        console.log(
-          `The image ${file.name} has been detected as inappropriate.`
-        );
-        return blurImage(file, BLURRED_BUCKET_NAME);
-      } else {
-        console.log(`The image ${file.name} has been detected as OK.`);
-      }
-    });
+  try {
+    const [result] = await client.safeSearchDetection(filePath);
+    const detections = result.safeSearchAnnotation || {};
+
+    if (
+      detections.adult === 'VERY_LIKELY' ||
+      detections.violence === 'VERY_LIKELY'
+    ) {
+      console.log(`The image ${file.name} has been detected as inappropriate.`);
```

Comment:
Concise, up-front context, and active tense makes for more easily scanned and understood log messages.

```suggestion
      console.log(`Detected image ${file.name} as inappropriate.`);
```